esbulk/esbulk.go

```diff
@@ -69,22 +99,32 @@ func NewIndexer(url string, index string, docs <-chan *estypes.Doc) *Indexer {
 
 			// Actually do the bulk insert once the buffer is full
 			if buf.Len() >= uploadat {
-				if err := upload(url, buf.Bytes()); err != nil {
-					indexer.err <- err
-					return
-				}
-
-				// Reset the buffer
-				buf.Reset()
+				wg.Add(1)
+				go func(b *bytes.Buffer) {
+					defer wg.Done()
+					if err := upload(targets[ti], b.Bytes()); err != nil {
+						indexer.err <- err
+						return
+					}
+
+					// Reset the buffer
+					b.Reset()
+					bufs <- b
+				}(buf)
+
+				buf = nil                    // go to next buffer in buffer pool
```

Comment:
Nope. No retries. The slightest hiccup is a hard fail. This is rotating
through a pool of buffers to allow parallel uploads.

As Eric pointed out it's not the most idiomatic way to parallelize things.
On Jan 20, 2016 4:51 PM, "Josh Roppo" <notifications@github.com> wrote:

> In esbulk/esbulk.go
> <https://github.com/lytics/escp/pull/12#discussion_r50345396>:
>
> > -				// Reset the buffer
> > -				buf.Reset()
> > +				wg.Add(1)
> > +				go func(b *bytes.Buffer) {
> > +					defer wg.Done()
> > +					if err := upload(targets[ti], b.Bytes()); err != nil {
> > +						indexer.err <- err
> > +						return
> > +					}
> > +
> > +					// Reset the buffer
> > +					b.Reset()
> > +					bufs <- b
> > +				}(buf)
> > +
> > +				buf = nil                    // go to next buffer in buffer pool
>
> This is a retry mechanism correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lytics/escp/pull/12/files#r50345396>.
>